contracts/access/AccessControl.sol

```diff
@@ -161,22 +160,13 @@ abstract contract AccessControl is Context {
      *
      * If `account` had not been already granted `role`, emits a {RoleGranted}
      * event.
-     */
-    function _grantRole(bytes32 role, address account) internal virtual {
-        if (_roles[role].members.add(account)) {
-            emit RoleGranted(role, account, _msgSender());
-        }
-    }
-
-    /**
-     * @dev Revokes `role` from `account`.
      *
-     * If `account` had been granted `role`, emits a {RoleRevoked} event.
+     * NOTE: Unlike {grantRole}, this function doesn't perform any checks on the
+     * calling account. Its usage should be restricted to grating the initial
+     * set of rules during contract construction.
      */
-    function _revokeRole(bytes32 role, address account) internal virtual {
-        if (_roles[role].members.remove(account)) {
-            emit RoleRevoked(role, account, _msgSender());
-        }
+    function _setupRole(bytes32 role, address account) internal virtual {
```

Comment:
We can restrict this function to work only in the constructor. See [`isConstructror`](https://github.com/OpenZeppelin/openzeppelin-sdk/blob/8bf20f7793ef86548519f8bda1d7eb6341013bbb/packages/lib/contracts/Initializable.sol#L47-L58). This is what I had in mind.

There is a caveat when initialization is done after construction, like an `initialize` function. However, in proxies the initializer runs in the constructor of the proxy as well. In cases other than proxies it should always be possible to have code in the constructor.